### PR TITLE
fdt: add fdt description document

### PR DESCRIPTION
**Pull request by leo11261**
## Summary
Add fdt description document to introdunce a libfdt modules and how to use it.
## Impact
help developers to know how to use libfdt in NuttX.

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR appears to be **incomplete** based on the NuttX requirements. 

**Here's why:**

* **Summary:**
    * Lacks details. What *exactly* does the fdt description document describe? What problem does it solve or what information does it provide?
* **Impact:**
    * While you marked everything as "no," adding documentation *does* have an impact. It improves understanding and usability for developers.  
* **Testing:**
    * Completely empty. How can you verify the document is helpful without any testing or validation?

**To improve this PR:**

1. **Expand the Summary:** Clearly articulate the purpose and content of the fdt description document.
2. **Revisit Impact:** Acknowledge the positive impact on documentation. Briefly describe how the document benefits users (e.g., "Provides a reference for understanding the device tree structure used by NuttX").
3. **Add Testing:** Even documentation needs validation. Describe how you reviewed the document for clarity, accuracy, and completeness. 

By addressing these points, you'll make your PR stronger and easier for reviewers to evaluate.

File: Documentation/guides/devicetree.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+====================
+Device Tree
+====================
+
+Overview
+--------
+
+Currently, Nuttx supports to parse FDT(Flattened Device Tree) using libfdt, a
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:8]:
```suggestion
Currently, NuttX supports parsing of FDT (Flattened Device Tree) using libfdt, a
```

Some typos.

File: Documentation/guides/devicetree.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+utility library for reading and manipulating the binary format:
+
+https://github.com/dgibson/dtc/
+
+Based on that, Nuttx has implemented some common functions to get properties.
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:13]:
```suggestion
Based on that, NuttX has implemented some common functions to get properties.
```

File: Documentation/guides/devicetree.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+And, Device tree's support in Nuttx is to reduce the configuration of chips/boards,
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:14]:
```suggestion
Device Tree support in NuttX will reduce the configuration of chips/boards,
```

File: Documentation/guides/devicetree.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+not used in nuttx kernel framework yet.
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:15]:
```suggestion
it is not used in NuttX kernel framework yet.
```

File: Documentation/guides/devicetree.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+How to use
+-----------
+
+1. Enable Device tree and libfdt
+
+Enable Kconfig
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:22]:
```suggestion
Enable the Kconfig settings:
```

File: Documentation/guides/devicetree.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    .. code-block:: console
+
+      CONFIG_DEVICE_TREE=y                        /* Enable Device Tree */
+      CONFIG_LIBFDT=y                             /* Enable utility library */
+
+2. Register DTB address
+
+Use fdt_register to let Nuttx know the dtb Address
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:31]:
```suggestion
Use fdt_register to set the DTB Address in NuttX
```

File: Documentation/guides/devicetree.rst [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+3. Parse DTB
+
+Chip/board use fdt_get to get dtb address, and then use fdt_* APIs to prase dtb
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:35]:
```suggestion
Chip/board will use fdt_get to get the DTB Address, and then use fdt_* APIs to parse the DTB
```

**Review by lupyuen — CHANGES_REQUESTED** [review]
Some typos to be fixed. Thanks!

File: Documentation/guides/devicetree.rst [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+1. Enable Device tree and libfdt
+
+Enable Kconfig
+
+    .. code-block:: console
+
+      CONFIG_DEVICE_TREE=y                        /* Enable Device Tree */
+      CONFIG_LIBFDT=y                             /* Enable utility library */
+
+2. Register DTB address
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:29]:
```suggestion
2. Register the DTB address
```

File: Documentation/guides/devicetree.rst [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+Use fdt_register to let Nuttx know the dtb Address
+
+3. Parse DTB
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:33]:
```suggestion
3. Parse the DTB
```

**Review by lupyuen — CHANGES_REQUESTED** [review]

File: Documentation/guides/devicetree.rst [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+https://github.com/dgibson/dtc/
+
+Based on that, Nuttx has implemented some common functions to get properties.
+And, Device tree's support in Nuttx is to reduce the configuration of chips/boards,
+not used in nuttx kernel framework yet.
+
+How to use
+-----------
+
+1. Enable Device tree and libfdt
```
Comment (lupyuen) [inline Documentation/guides/devicetree.rst:20]:
```suggestion
1. Enable Device Tree and libfdt
```

**Review by lupyuen — CHANGES_REQUESTED** [review]

**Review by lupyuen — APPROVED** [review]
Looks great. Thank you so much :-)

**Pull request merged**